src/zenml/models/model_models.py

```diff
@@ -150,59 +151,120 @@ def pipeline_runs(self) -> Dict[str, PipelineRunResponseModel]:
             for name, pr in self.pipeline_run_ids.items()
         }
 
-    def get_model_object(
-        self, name: str, version: Optional[str] = None
-    ) -> ArtifactResponseModel:
+    def _get_linked_object(
+        self,
+        collection: Dict[str, Dict[str, UUID]],
+        name: str,
+        version: Optional[str] = None,
+        pipeline_name: Optional[str] = None,
+        step_name: Optional[str] = None,
+    ) -> Optional[ArtifactResponseModel]:
         """Get model object linked to this model version.
 
         Args:
+            collection: The collection to search in.
```

Comment:
This needs more explanation - what kind of collection? What am I supposed to pass in here?

Reply:
This is internal method, as underscore shows - you shouldn't use it anyhow directly 🙂 
But I can improve docstring here.

Reply:
Rephrased